Spigot-Server-Patches/0696-Add-EntityExtinguishEvent.patch

```diff
@@ -0,0 +1,210 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Aurora <aurora@relanet.eu>
+Date: Sat, 20 Mar 2021 18:14:43 +0100
+Subject: [PATCH] Add EntityExtinguishEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/Entity.java b/src/main/java/net/minecraft/world/entity/Entity.java
+index 429f0591c6a55f6c5d08a0755f7d39da676468bc..86c4fdde7d282a20d5889d9f5f43048893eb8ed1 100644
+--- a/src/main/java/net/minecraft/world/entity/Entity.java
++++ b/src/main/java/net/minecraft/world/entity/Entity.java
+@@ -3,6 +3,7 @@ package net.minecraft.world.entity;
+ import com.google.common.collect.Iterables;
+ import com.google.common.collect.Lists;
+ import com.google.common.collect.Sets;
++import io.papermc.paper.event.entity.EntityExtinguishEvent;
+ import it.unimi.dsi.fastutil.objects.Object2DoubleArrayMap;
+ import it.unimi.dsi.fastutil.objects.Object2DoubleMap;
+ import java.util.Arrays;
+@@ -529,12 +530,22 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, ne
+         this.m();
+         this.aJ();
+         if (this.world.isClientSide) {
++            // Paper start - add EntityExtinguishEvent
++            EntityExtinguishEvent event = CraftEventFactory.callEntityExtinguishEvent(this, EntityExtinguishEvent.Cause.TIME);
++            if (!event.isCancelled()) {
+             this.extinguish();
++            }
++            // Paper end
```

Comment:
Kind of nitpick: that code you added is never used, so why add it? (It is never used since the server is not client-side.)